examples/source/interactivity-dynamic-content/Dynamic_Menu_With_Sub_Menus.html

```diff
@@ -0,0 +1,167 @@
+    #sideNav {
+      width: 100%;
+      height: 100%;
+      background-color: #fff;
+      max-width: 100vw;
+    }
+
+    .header-primary li a {
+      padding: 8px 8px 8px 32px;
+      text-decoration: none;
+      font-size: 16px;
+      color: #818181;
+      display: block;
+      transition: 0.3s;
+      border-bottom: 1px solid #e6e6e6;
+      font-family: "Open Sans", sans-serif;
+      font-size: 16px;
+      font-weight: 400;
+      line-height: 24px;
+      outline: 0;
+    }
+
+    .main-nav {
+      width: 100%;
+      height: 100%;
+      background-color: #fff;
+      max-width: 100vw;
+    }
+
+    .mobile-menu-header {
+      background-color: blue;
+      height: 45px;
+      text-align: center;
+      line-height: 45px;
+      color: white;
+    }
+
+    .closebtn {
+      top: 0px;
+      right: 15px;
+      outline: 0;
+      position: absolute;
+    }
+
+    .innermenulinks {
+      padding: 8px 8px 8px 32px;
+      text-decoration: none;
+      font-size: 16px;
+      color: #818181;
+      display: block;
+      transition: 0.3s;
+      border-bottom: 1px solid #e6e6e6;
+      font-family: "Open Sans", sans-serif;
+      line-height: 24px;
+      font-weight: 400;
+    }
+
+    .rightarrow:after {
+      content: ">";
+      float: right;
+      padding-right: 10px;
+      font-weight: bolder;
+      font-size: 25px;
+    }
+
+    .leftarrow:before {
+      content: "<";
+      float: left;
+      padding-left: 10px;
+      font-weight: bolder;
+      font-size: 25px;
+    }
+
+  </style>
+</head>
+<!-- ## Implementation -->
+  <!--
+      We use two `amp-sidebar` elements nested within eachother.
+      
+      `amp-list` is utilized to retrieve a formatted JSON describing the menu and submenus.
+
+      `amp-mustache` is used for templating. {{items}} and {{submenu}} check for the presence of the keys in the JSON object.      
+      
+  -->
+<body>
+  
+  <header>
+    <h3>
+      Dynamic Sidebar
+    </h3>
+    <p>
+      This side bar is dynamically constructed from a JSON response, which contains a top level and a sub level
+    </p>
+    <button aria-label="open sidebar" on="tap:sideNav.toggle" tabindex="0">
+      Toggle Sidebar
+    </button>
+  </header>
+
+  <amp-sidebar id="sideNav" layout="nodisplay" side="left">
+    <div class="mobile-menu-header">
+      <span>Menu</span >
+      <i class="closebtn" on="tap:sideNav.close" tabindex="0" role="on">×</i>
+    </div>
+    <ul class="header-primary">
+      <amp-list width="auto" height="435" layout="fixed-height" src="/static/samples/json/menu.json" single-item
+        items="." id="nav-list">
+        <template type="amp-mustache" id="amp-template-id">
+          {{#items}}
+          <li>
+            <a class="rightarrow"
+              role="button" 
+              on="tap:{{title}}_Slide.toggle;"
+              tabindex="0">{{title}}</a>
+          </li>
+          <amp-sidebar id="{{title}}_Slide" layout="nodisplay" class="main-nav" side="right">
```

Comment:
Wouldn't  amp-nested-menu work. better here?`https://amp.dev/documentation/components/amp-sidebar/#building-nested-menus

Reply:
I'm not sure... I actually tried it out here: https://glitch.com/edit/#!/amp-snippets?path=dynamic_menu_sub_menu/index.html

When you click into a submenu, the transition occurs but nothing appears, despite the li elements all being populated with the correct info when you look at the DOM. The inspector even highlights the boundaries of these elements when you select them in the elements tab.

I can't figure out why they aren't appearing: do you mind taking a look? Otherwise, this would be the preferred method, but I'm not sure if this is an amp-list + amp-nested-menu interaction bug.

Reply:
Looks like a bug to me. Can you please file a report.

Reply:
Drive by comment: `<amp-nested-menu>` works when the entire `<amp-nested-menu>` is inside the `<amp-list>` template. See the documentation demo here: https://github.com/ampproject/amphtml/blob/master/extensions/amp-nested-menu/amp-nested-menu.md#dynamic-content-rendering

=) 

I think we can probably be more explicit about that in the documentation.